Docs for backwards compatibility (#763)
nsidc/earthaccess: docs/user_guide/backwards-compatibility.md

    1. There are 3 version parts: MAJOR.MINOR.PATCH (e.g. 1.2.3 is major version 1, minor release 2, patch release 3).
    2. When the major version changes, anything can break! Always visit this documentation before doing a major upgrade.
    3. When the minor version changes, new features should be available, but nothing should break. Visit the documentation to learn about new features.
    4. When the patch version changes, only bugfixes should be included. Visit the CHANGELOG to learn more about the fixes.

## Our commitments

### Versioning

We will follow SemVer. All version changes will consider [this](https://example.com/our-public-api) to be the public API documentation for the purposes of deciding whether a change is breaking or non-breaking.

### Release Communication

1. We will announce releases on the following channels: ___
2. Release announcements will include a prominent notification of breaking changes, including a link to migration guide.

### CHANGELOG

1. We will update the CHANGELOG for every release
2. The CHANGELOG will include prominent notification of breaking changes, including a link to migration guide.

### Fixing Backwards Incompatible Changes

> 1. We will plan to fix any backwards incompatible changes in non-major releases.

Reviewer: ```suggestion
1. We will plan to fix any backwards incompatible changes in non-major releases _or_ re-release with a major version bump.
```

I don't think we can necessarily commit to always doing this, sometimes it might be a mistake. I'm not sure. What's everyone else thing?

> 2. We cannot guarantee developers will be available to complete this work alongside other priorities.

Reviewer: ```suggestion
2. We cannot guarantee a timeline under which maintainers will be able to complete this work alongside their other priorities.
```

3. Our maintenance team will always welcome outside contributions towards this goal.

> 4. Please use an issue to communicate about this work.

Reviewer: ```suggestion
4. Please use a [GitHub Issue](https://github.com/nsidc/earthaccess/issues) to communicate about this work.
```

### Release Communication

1. We will announce releases on the following channels: ___
2. Release announcements will include a prominent notification of breaking changes, including migration instructions.

### CHANGELOG

1. We will update the CHANGELOG for every release following [Common Changelog](https://common-changelog.org/) style.
2. The CHANGELOG will include prominent notification of breaking changes, including a link to migration guide.

### Fixing Backwards Incompatible Changes

1. We will plan to fix any backwards incompatible changes in non-major releases _or_ re-release with a major version bump.

2. We cannot guarantee a timeline under which maintainers will be able to complete this work alongside their other priorities.

3. Our maintenance team will always welcome outside contributions towards this goal.

4. Please use a [GitHub Issue](https://github.com/nsidc/earthaccess/issues) to communicate about this work.

## Our Python and dependency support policy

> Our project follows the SPEC0 dependency deprecation policy, which outlines the guidelines for deprecating and removing dependencies from our codebase.

Reviewer: ```suggestion
Our project follows the [SPEC0](https://scientific-python.org/specs/spec-0000/) dependency version deprecation policy, which outlines the guidelines for upgrading dependencies of _earthaccess_.
```
